addon/components/horizontal-bar-chart.js

```diff
@@ -431,37 +440,112 @@ const HorizontalBarChartComponent = ChartComponent.extend(FloatingTooltipMixin,
+    }
+  },
 
-      const positiveGroupingLabels = positiveValues.map((val) => {
-        return this._getElementForValue(groupLabelElements, val);
-      });
-      const positiveValueLabels = positiveValues.map((val) => {
-        return this._getElementForValue(valueLabelElements, val);
-      });
-      const negativeGroupingLabels = negativeValues.map((val) => {
-        return this._getElementForValue(groupLabelElements, val);
-      });
-      const negativeValueLabels = negativeValues.map((val) => {
-        return this._getElementForValue(valueLabelElements, val);
-      });
+  /*
+   * Determine the label widths on either side of a chart which contains a mix of positive
+   * and negative values
+   * @private
+   * @param {Array.<SVGTextElement>} groupLabelElements The text elements
+   *  representing the group labels for the chart
+   * @param {Array.<SVGTextElement>} valueLabelElements The text elements
+   *  representing the value labels for the chart
+   * @return {Object.<String, Number>}
+   */
+  _computeMixedLabelWidths: function(groupLabelElements, valueLabelElements) {
+    const minValue = this.get('minValue');
+    const maxValue = this.get('maxValue');
+    const maxLabelWidth = this.get('maxLabelWidth');
 
-      const leftLabels = negativeValueLabels.concat(positiveGroupingLabels);
-      const rightLabels = positiveValueLabels.concat(negativeGroupingLabels);
+    // The grouping labels for positive values appear on the left side of the chart axis, and
+    // vice-versa for negative values and right labels
+    const leftGroupingLabels = this.get('positiveValues').map((val) => {
+      return this._getElementForValue(groupLabelElements, val);
+    });
+    const rightGroupingLabels = this.get('negativeValues').map((val) => {
+      return this._getElementForValue(groupLabelElements, val);
+    });
+    const maxLeftGroupingLabelWidth = d3.min([maxLabelWidth,
+      this._maxWidthOfElements(leftGroupingLabels)]);
+    const maxRightGroupingLabelWidth = d3.min([maxLabelWidth,
+      this._maxWidthOfElements(rightGroupingLabels)]);
+
+    // The value label that is furthest to the left is the one representing the minimum
+    // value in the chart, and vice-versa for the right side and maximum value
+    const leftMostValueLabelWidth = this._getElementWidthForValue(valueLabelElements, minValue);
+    const rightMostValueLabelWidth = this._getElementWidthForValue(valueLabelElements, maxValue);
+
+    const padding = (2 * this.get('labelPadding')) + this.get('yAxisTitleHeightOffset') + 5;
+    const outerWidth = this.get('outerWidth');
+
+    const width = outerWidth - leftMostValueLabelWidth - rightMostValueLabelWidth - padding;
+    const xScale = this._xScaleForWidth(width);
+
+    const maxNegativeBarWidth = Math.abs(xScale(minValue) - xScale(0));
+    const maxPositiveBarWidth = Math.abs(xScale(maxValue) - xScale(0));
+
+    // If the sum of the widths of the longest bar in a direction and its value label is larger
```

Comment:
where is the if inside the code reflecting the if in this comment? Why is this comment put on this variable?